### PR TITLE
Decouple title, header and footer components from jsp pages

### PR DESCRIPTION
**Pull request by tharikaGitHub**
### Proposed changes in this pull request
This PR decouples the header, footer and title components from the following webapps.

1. authenticationendpoint
2. accountrecoveryendpoint

### CODE REVIEW

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/src/main/webapp/consent.jsp [hunk 1]
```diff
@@ -39,8 +40,15 @@
     <meta http-equiv="X-UA-Compatible" content="IE=edge">
     <meta charset="utf-8">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
-    <title><%=AuthenticationEndpointUtil.i18n(resourceBundle, "wso2.identity.server")%>
-    </title>
+    <!-- title -->
+    <%
```
Comment (jeradrutnam) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/src/main/webapp/consent.jsp]: Shall we move the logic to an util method and resolve everywhere?

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/src/main/webapp/consent.jsp [hunk 2]
```diff
@@ -20,6 +20,7 @@
 <%@ page import="org.wso2.carbon.identity.application.authentication.endpoint.util.Constants" %>
 <%@include file="localize.jsp" %>
 <%@include file="init-url.jsp" %>
+<%@ page import="java.io.File"%>
```
Comment (jeradrutnam) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/src/main/webapp/consent.jsp]: Move page import before includes on all the pages

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/src/main/webapp/consent.jsp [hunk 3]
```diff
@@ -39,8 +40,15 @@
...
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
-    <title><%=AuthenticationEndpointUtil.i18n(resourceBundle, "wso2.identity.server")%>
-    </title>
+    <!-- title -->
+    <%
+        File titleFile = new File(getServletContext().getRealPath("extensions/title.jsp"));
+        if (titleFile.exists()) {
+    %>
+            <jsp:include page="extensions/title.jsp"/>
+    <%} else {%>
```
Comment (jeradrutnam) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/src/main/webapp/consent.jsp]: `<%} else {%>` keep a space in between `<% %>`. E.g. <% } else { %>

Do it on all the pages

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/src/main/webapp/consent.jsp (hunk 2 above)
Comment (tharikaGitHub) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/src/main/webapp/consent.jsp]: Fixed with https://github.com/wso2/carbon-identity-framework/pull/2426/commits/37e4813fd5f8e9078546c990764edd8d9132b13e

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/src/main/webapp/consent.jsp (hunk 3 above)
Comment (tharikaGitHub) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/src/main/webapp/consent.jsp]: Fixed with https://github.com/wso2/carbon-identity-framework/pull/2426/commits/37e4813fd5f8e9078546c990764edd8d9132b13e

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/src/main/webapp/consent.jsp (hunk 1 above)
Comment (jeradrutnam) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/src/main/webapp/consent.jsp]: Will do it as a post enhancement

**Pull request merged**